plugins/optimization-detective/class-od-url-metric.php

```diff
@@ -417,6 +425,18 @@ public function get_uuid(): string {
 		return $this->data['uuid'];
 	}
 
+	/**
+	 * Gets ETag.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return string ETag.
+	 */
+	public function get_etag(): string {
+		// Since the ETag is optional for now, return an empty string for old URL metrics that do not have one.
+		return $this->data['eTag'] ?? '';
```

Comment:
```suggestion
	 * @return string|null ETag.
	 */
	public function get_etag(): ?string {
		// Since the ETag is optional for now, return null for old URL Metrics that do not have one.
		return $this->data['eTag'] ?? null;
```

Reply:
This follows the pattern of other getters which return `null` when the data is absent. In a future release when we make `etag` required, then we can make it always return `string`.